crates/uv/tests/it/lock.rs

```diff
@@ -3867,7 +3867,7 @@ fn lock_conflicting_mixed() -> Result<()> {
 
         [package.dev-dependencies]
         project1 = [
-            { name = "sortedcontainers", version = "2.3.0", source = { registry = "https://pypi.org/simple" } },
+            { name = "sortedcontainers", version = "2.3.0", source = { registry = "https://pypi.org/simple" }, conflict-marker = "extra != 'project2'" },
```

Comment:
Encoding the names in the string value would be an internal detail _not_ exposed to the lock file. It would be written like it is here, and when groups are added, it would be `group != 'foobar'`.

(My three choices given on Discord all have the same behavior with respect to the lock file and how it's written there. The only differences between them are how we implement it.)

Reply:
Oh ok, ack.

Reply:
To follow up here: when I wrote the above comment, I was still thinking we'd have "normal PEP 508 marker" and "conflict marker" as separate things in the lock file. And in that scenario, it seemed more justifiable to have our own little custom syntax for the conflict marker that we mapped back to PEP 508 markers _internally_. But since then, I've come to understand that these markers need to be combined since the conflict markers themselves can be dependent on the PEP 508 markers.

I didn't want to make the markers no longer parseable by PEP 508... Maybe that's something we should do in the future, but I felt it was good not to break that property at least for now. But that does mean that we end up with some funny looking markers that I was originally trying to avoid. (The comments and commit messages give more detail.)